src/service-api.ts

```diff
@@ -19,24 +20,68 @@ const plugin: FastifyPluginAsync = async (fastify) => {
   // get current user
   fastify.get('/current', async ({ member }) => member);
 
-    // // get member
-    // fastify.get(
-    //   '/:id',
-    //   { schema: getOne },
-    //   async ({ category, id , log }) => {
-    //     const task = taskManager.createGetTask(category, id);
-    //     return runner.runSingle(task, log);
-    //   },
-    // );
-
-    // get members
-    fastify.get(
-      '/allcategories',
-      async ({ member, log }) => {
-        const task = taskManager.createGetAllTask(member);
+    // get category info
+  fastify.get<{ Params: {categoryId: string };}>(
+    '/category/info/:categoryId',
+    async ({ member, params: {categoryId}, log }) => {
+      const task = taskManager.createGetCategoryTask(member, categoryId);
+      return runner.runSingle(task, log);
+    },
+  );
+
+  // get types
+  fastify.get(
+    '/category/types',
+    async ({ member, log }) => {
+      const task = taskManager.createGetCategoryTypesTask(member);
+      return runner.runSingle(task, log);
+    },
+  );
+
+  // get categories of given type
+  fastify.get<{ Params: {typeId: string}; }>(
+    '/categories/:typeId',
```

Comment:
If you go with `?typeId` you can drop it, and this can also be used to get all categories of all types.